controllers/templatesync/template_sync.go

```diff
@@ -1693,27 +1693,27 @@ func (r *PolicyReconciler) setCreatedGkConstraint(b bool) {
 	r.createdGkConstraint = &b
 }
 
-// Check duplicate names in policy-templates(configurationPolicies)
-func hasDupName(pol *policiesv1.Policy) bool {
+// Check duplicate names in policy-templates and return the first duplicate name found, if any.
+func getDupName(pol *policiesv1.Policy) string {
 	templates := pol.Spec.PolicyTemplates
 
-	foundNames := make(map[string]struct{})
+	foundNames := make(map[string]bool, len(templates))
 
 	for _, v := range templates {
 		unstructured, err := unmarshalFromJSON(v.ObjectDefinition.Raw)
 		if err != nil {
 			// Skip unmarshal error here, template error should appear later
-			return false
+			return ""
 		}
 
 		name := unstructured.GetName()
 
-		if _, has := foundNames[name]; has {
-			return true
+		if has := foundNames[name]; has {
```

Comment:
Shouldn't this check the value here rather than assigning it?
```suggestion
		if foundNames[name] {
```

Reply:
It does the same thing since `has` is the boolean value but your suggestion is more concise.